src/h5web/visualizations/heatmap/hooks.ts

```diff
@@ -81,5 +81,14 @@ export function useTextureData(): TextureDataState {
     values,
   ]);
 
+  const [rows, cols] = useDims();
+  const dims = `${rows}x${cols}`;
+  const [prevDims, setPrevDims] = useState<string>();
+
+  if (prevDims !== dims) {
+    setPrevDims(dims);
+    mergeState({ textureData: undefined });
+  }
```

Comment:
Another great use case for derived state ^^ -- and it's all self-contained within the existing custom hook.

Reply:
I have a minor complain about this: the textureData is not reset when the dimensions of the new dataset are the ones of the old dataset (ex: swapping dimensions of a square matrix). This poses no display problem as the texture stays the same as the dimensions are the same but still, we get a different behaviour from when the dimensions are different as the texture is set to gray in the latter case. Perhaps we could attach some `id` to the `dataArray` and reset the texture on this ?

This being said, your implementation covers 99% of the cases and for the remaining 1%, often the rendering is so quick that we don't notice anything...

Reply:
Yeah, it's on purpose. The texture loader gives a good-enough indication that something is about to change, so the fact that the texture underneath is stale is not a problem IMO. It's actually consistent with the change of color map/domain. Also, it saves a render.

If I had wanted to the clear the texture every time, I would have checked `dataArray.shape` directly as we get a new reference each time the mapping changes.

Reply:
Makes sense.